### ISSUE TITLE
[Feature Request]: Assets by screen density

### ISSUE THREAD

**Issue by binouze** [google/play-unity-plugins#2]
Hi, 
As I understand in the Android App Bundle documentation a split is done on the res/drawable-{density} folders.
It would be great to add an option to be able to choose assets to be packaged by screen density.

For exemple, I'm making 2D games and I use Sprite Atlases, it would be so nice to be able to choose Sprite Atlases variant by screen densities to deliver smallest APK on small phones.

**Comment by paulohgodinho** [comment]
I believe you can pack your atlases in multiple Asset Bundles and download the correct ones from google CDN based on the screen pixel density.

**Comment by binouze** [comment]
Yes sure but it would be good to have the same option as iOS app slicing for android too. Without the need of downloading assets at first game launch.

**Comment by thanwah** [comment]
Unity doesn't make much use of traditional Android resources, but automatically targeting Sprite Atlases based on the device screen size or density is an interesting thought! We have been working on targeting assets based on device attributes, but I'm not sure whether Sprite Atlases came up yet.

It looks like it might be possible to automatically create [Variant Sprite Atlases](https://docs.unity3d.com/Manual/VariantSpriteAtlas.html) with a scale factor that corresponds to a particular screen size and/or density.

I'll bring this up with a team member who has been working on device-specific asset targeting. Thank you for filing the feature request!

**Comment by binouze** [comment]
Would be very nice :)

**Comment by binouze** [comment]
After reading the code of AssetDelivery plugin, I see that there is a naming convention for texture compression format targeting **is there a naming convention for device screen density ?**

At the moment I use assets delivery on demand with 4 AssetBundle (just containing SpriteAtlas) (hd_astc, sd_astc, hd_etc2, sd_etc2) and at first game launch I get the right bundle for the target screen size and support of texture. 

The first launch is the most important for our users and I would like it to be fast, not let the user wait while downloading assets. Having a split on texture support and screen size would be a must.

About "automatically create Variant Sprite Atlases", personally I prefer keep control on sprite atlas variants and for all our projects at the moment 2 size variants are really enough.

An other problem, it seems that there is a problem with SpriteAtlas variants packed in AssetBundle, I just posted on unity forum about it here : [https://forum.unity.com/threads/spriteatlas-variant-in-assetbundle-references-master-spriteatlas-as-dependency.875350/](url)

**Comment by thanwah** [comment]
We don't have a naming convention yet for device screen density or size.

@4ian and I have started talking about this feature request, but we don't have any updates yet.

Please keep us in the loop on your Unity forum question since it seems like that may be a blocker for the approach to work.

Thanks!

**Comment by binouze** [comment]
No response to my question on the forum. I filled a bug report this morning and I'm waiting for an update on this.

**Comment by binouze** [comment]
Hi, 

I managed to don't have the master SpriteAtlas packed in the AssetBundle containing the variant SpriteAtlas when the master SpriteAtlas is not in any AssetBundle, just unchecked the "include in build" tick in the SpriteAtlas.

But the dependency is still there in the dependencies list of the AssetBundles manifests when the master SpriteAtlas is packed in an AssetBundle.

In your system, does the dependencies are automatically loaded alongside the demanded resource ?

**Comment by thanwah** [comment]
Thanks for following up!

We check AssetBundle dependencies and display them in the settings UI, however we don't enforce that AssetBundle dependencies are satisfied when downloading asset packs. (One reason we don't enforce dependencies is that Unity allows circular dependencies among AssetBundles.)

In your case, if the AssetBundle containing the variant SpriteAtlas depends on the AssetBundle containing the master SpriteAtlas, this should be fine, unless the variant SpriteAtlas asset itself requires the master SpriteAtlas asset to be loaded. (I'm not sure if this is the case, but you should be able to test out loading your variant SpriteAtlas to check.)

**Comment by binouze** [comment]
I already tested and the variant sprite atlas is working even if the master is not loaded. 
It would be weird to have to load the master atlas to be able to use the variant ;)

**Comment by binouze** [comment]
Hi,
I'm currently using asset delivery for different texture format (#tcf_default and #tcf_astc) but I still need 2 different AAB with manifest tweaking to support HD / SD versions. There is no more issues with SpriteAtlas variants in AssetsBundles and the texture format delivery works great.
Would be good to add the screen density splitting support ;)

**Comment by binouze** [comment]
Hi there,
Any update on screen density / size splitting support ?

**Comment by lvgrecords** [comment]
Hi there,
Any update on screen density / size splitting support ?
